hadoop-plugin/src/main/groovy/com/linkedin/gradle/hadoopdsl/checker/RequiredFieldsChecker.groovy

```diff
@@ -242,6 +243,32 @@ class RequiredFieldsChecker extends BaseStaticChecker {
     foundError |= validateNotEmpty(job, "errorHdfsPath", job.errorHdfsPath);
   }
 
+  @Override
+  void visitJob(SqlJob job) {
+    foundError |= validateNotEmpty(job, "jdbcDriverClass", job.jdbcDriverClass);
+    foundError |= validateNotEmpty(job, "jdbcUrl", job.jdbcUrl);
+    foundError |= validateNotEmpty(job, "jdbcUserId", job.jdbcUserId);
+    foundError |= validateNotEmpty(job, "jdbcEncryptedCredential", job.jdbcEncryptedCredential);
+    foundError |= validateNotEmpty(job, "jdbcCryptoKeyPath", job.jdbcCryptoKeyPath);
+  }
+
+  private boolean validateTeradataCredential(Job job,
+                                             String credentialName,
+                                             String encryptedCredential,
+                                             String cryptoKeyFilePath) {
+    boolean isCredentialExist = false;
+    isCredentialExist ^= credentialName == null || credentialName.isEmpty() == null;
+    isCredentialExist ^= encryptedCredential == null || encryptedCredential.isEmpty() == null;
```

Comment:
So Teradata jobs must have EITHER credentialName OR encryptedCredential, BUT NOT BOTH. That's what your code says. Is that correct?